tools/noms/staging.py

```diff
@@ -28,36 +30,76 @@ def Main(projectName, stagingFunction):
         staging.Main('nerdosphere', staging.GlobCopier('index.html', 'styles.css', '*.js'))
     """
     parser = argparse.ArgumentParser(description='Stage build products from this directory.')
-    parser.add_argument('stagingDir',
+    parser.add_argument('staging_dir',
                         metavar='path/to/staging/directory',
                         type=_dir_path,
                         help='top-level dir into which project build products are staged')
     args = parser.parse_args()
-    projectStagingDir = os.path.join(args.stagingDir, projectName)
+    project_staging_dir = os.path.join(args.staging_dir, projectName)
 
-    normalized = os.path.realpath(projectStagingDir)
-    if not _is_sub_dir(projectStagingDir, args.stagingDir):
-        raise Exception(projectStagingDir + ' must be a subdir of ' + args.stagingDir)
+    normalized = os.path.realpath(project_staging_dir)
+    if not _is_sub_dir(project_staging_dir, args.staging_dir):
+        raise Exception(project_staging_dir + ' must be a subdir of ' + args.staging_dir)
 
     if not os.path.exists(normalized):
         os.makedirs(normalized)
     stagingFunction(normalized)
 
 
-def GlobCopier(*globs):
+def run_globs(staging_dir, globs, exclude):
+    for pattern in globs:
+        for f in glob.glob(pattern):
+            if os.path.isdir(f):
+                continue
+            from_dir, name = os.path.split(f)
+            if name in exclude:
+                continue
+            to_dir = os.path.join(staging_dir, from_dir)
+            if not os.path.exists(to_dir):
+                os.makedirs(to_dir)
+            yield (f, to_dir)
+
+
+def rename_with_hash(f, to_dir, rename_dict):
+    digest = ''
```

Comment:
Is this a pylint thing? Because python scoping rules make the assignment on line 68 perfectly happy.